#!/usr/bin/env python
PACKAGE="neo_ros_pc2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("median_filter_", bool_t, 0, "Using median filter", True)
gen.add("median_filter_half_windows_size_", int_t, 0, "Median filter windows size(odd)", 1, 1, 5 )

gen.add("close_point_filter_", bool_t, 0, "Using close point filter", True)
gen.add("close_point_distance_", int_t, 0, "Close Point parameter", 50, 1, 100)
gen.add("max_distance_", int_t, 0, "Max distance", 4000, 1000, 5000)

gen.add("largest_square_distance_of_line_", double_t, 0, "Max square distance of a line ", 400, 100, 900)
gen.add("collinearity_param_", double_t, 0, "Collinearity parameter ", 2000, 500, 5000)
gen.add("interpolation_point_num_", int_t, 0, "Interpolation point number ", 1, 0, 5)
gen.add("num_points_of_line_", int_t, 0, "Min number points of a line ", 5, 4, 10)

exit(gen.generate(PACKAGE, "neo_ros_pc2", "Filter"))
